### PR TITLE
Create get rooms endpoint

### LINKED ISSUE #57: Implement get all rooms endpoint

### PR DESCRIPTION
**Pull request by FabianAronsson**
Adds a get all rooms endpoint. This is for an administrator only since otherwise you could leak other peoples room ids.

The endpoint also has support for filtering based on ids and room names.

The api specification will be updated.

Closes #57

### CODE REVIEW

File: Queueomatic/Server/Endpoints/Room/GetAll/GetAllRoomsEndpoint.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    public override void Configure()
+    {
+        Get("/rooms");
+        Roles("Administrator");
+    }
+
+    public override async Task HandleAsync(GetAllRoomRequest req, CancellationToken ct)
+    {
+        var roomId = _hashIdService.Decode(req.RoomId);
```
Comment (Mojtabarezaei4) [inline Queueomatic/Server/Endpoints/Room/GetAll/GetAllRoomsEndpoint.cs:29]: What will be the response of the endpoint when roomId gets -1 as value?

Comment (FabianAronsson) [reply Queueomatic/Server/Endpoints/Room/GetAll/GetAllRoomsEndpoint.cs:29]: If you check the repository file you will see that the filter handles any negative numbers.

Comment (FabianAronsson) [reply Queueomatic/Server/Endpoints/Room/GetAll/GetAllRoomsEndpoint.cs:29]: Yes it is. For example if you decide to omit the query parameter and search only by name, you would want it to return those results.

Comment (Mojtabarezaei4) [reply Queueomatic/Server/Endpoints/Room/GetAll/GetAllRoomsEndpoint.cs:29]: Okay. Makes sense.

**Review by Mojtabarezaei4 — APPROVED** [review]

**Pull request merged**